tests/project/utils.ts

```diff
@@ -1,25 +1,54 @@
-import type {ForkOptions} from 'child_process';
-import {fork} from 'child_process';
+import type {Readable} from 'stream';
+import {PassThrough} from 'stream';
+import type {ChildProcess} from 'child_process';
+import {once} from 'events';
+import getStream from 'get-stream';
+import {promiseWithTimeout} from '../support/test-utils';
 
-export interface ResolvedChildProcess {
-    code: number;
-    signal: NodeJS.Signals;
-    stdout: string;
-    stderr: string;
+export type ChildClosedOptions = getStream.OptionsWithEncoding & { serialization?: false | 'json'; timeout?: number };
+
+export type ChildClosedPayload = { stdout: string; stderr: string; response?: any };
+
+/**
+ * Returns a promise that resolves with a child process's close event
+ * Also handles character set decoding, deserialization from JSON, accepts a timeout in ms
+ *
+ * @example
+ * const child = fork('./script.js');
+ * const { stdout, stderr, response } = await watchChild(child);
+ * const { exitCode } = child;
+ */
+export async function childClosed(child: ChildProcess, options?: ChildClosedOptions): Promise<ChildClosedPayload> {
+    const deserialize = (output: string) => options?.serialization === 'json' ? JSON.parse(output) : output;
+    const collect = (readable: Readable) => getStream(readable || new PassThrough(), options);
+    const childPromises = Promise.all([once(child, 'close'), collect(child.stdout), collect(child.stderr)])
+        .then(([, stdout, stderr]) => ({stdout, stderr, response: deserialize(stdout)}));
+    if (options?.timeout) {
+        return promiseWithTimeout(options.timeout, childPromises);
+    }
+    return childPromises;
 }
 
-export async function forka(modulePath: string, args?: readonly string[], options?: ForkOptions): Promise<ResolvedChildProcess> {
-    return new Promise((resolve) => {
-        const proc = fork(modulePath, args, {silent: true, ...options});
-        let stdout = '', stderr = '';
-        proc.stderr.on('data', (chunk) => {
-            stderr += chunk.toString();
-        });
-        proc.stdout.on('data', (chunk) => {
-            stdout += chunk.toString();
-        });
-        proc.on('close', (code, signal) => {
-            resolve({code, signal, stdout: stdout || undefined, stderr: stderr || undefined});
+/**
+ * Returns a promise that resolves when a certain condition has been met, based on the content of a stream.
+ *
+ * @example
+ * await watchStream(child.stdout).forMatch(/Config loaded/)
+ * thisRunsAfterConfigLoads();
+ */
+export const watchStream = (readable: Readable, options?: { encoding: BufferEncoding }) => {
```

Comment:
Not sure why TS isn't erroring about this:
```suggestion
export const watchStream = (readable: Readable, options?: {encoding: BufferEncoding}) => {
```

Reply:
I'll check eslint.

Reply:
Looks like two things are happening: 

1. The rule (https://eslint.org/docs/rules/object-curly-spacing) doesn't apply to this type of bracket. I put this into a file under `src/`:

![Screenshot_2022-03-08_05-32](https://user-images.githubusercontent.com/250466/157230413-9fe88e70-a0e5-4d4b-85ee-dcd009257140.png)

2. The typescript test files aren't being included in the ESlint scan unless their `tsconfig.json` is explicitly referenced in `eslintrc.js` under `*.parserOptions.project`. I'm really not sure about how to put them all in there (like if there's a way to do a wildcard or something). For now, I'll add `tests/project/tsconfig.json` here. I'll add the rest of the projects to https://github.com/darkreader/darkreader/pull/8210.

Reply:
#2 was a false alarm.. Only #1 is an issue, from what I'm discovering.